bmq-sdk/src/test/java/com/bloomberg/bmq/resultcodes/AckResultTest.java

```diff
@@ -0,0 +1,157 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.bloomberg.bmq.resultcodes;
+
+import static com.bloomberg.bmq.resultcodes.AckResultChecks.checkGeneric;
+import static com.bloomberg.bmq.resultcodes.AckResultChecks.checkLimitBytes;
+import static com.bloomberg.bmq.resultcodes.AckResultChecks.checkLimitMessages;
+import static com.bloomberg.bmq.resultcodes.AckResultChecks.checkStorageFailure;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkCanceled;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkIfNotGeneric;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkInvalidArgument;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkNotConnected;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkNotReady;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkNotSupported;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkRefused;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkSuccess;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkTimeout;
+import static com.bloomberg.bmq.resultcodes.GenericResultChecks.checkUnknown;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.bloomberg.bmq.ResultCodes;
+import org.junit.jupiter.api.Test;
+
+// ================================================================
+// AckResult tests
+// ================================================================
+public class AckResultTest {
+    public static class AckGenericTest {
+        public void isLimitMessages(ResultCodes.AckCode obj) {
+            assertFalse(obj.isLimitMessages());
+        }
+
+        public void isLimitBytes(ResultCodes.AckCode obj) {
+            assertFalse(obj.isLimitBytes());
+        }
+
+        public void isStorageFailure(ResultCodes.AckCode obj) {
+            assertFalse(obj.isStorageFailure());
+        }
+    }
+
+    public static class LimitMessagesTest extends AckGenericTest {
+        @Override
+        public void isLimitMessages(ResultCodes.AckCode obj) {
+            assertTrue(obj.isLimitMessages());
+        }
+    }
+
+    public static class LimitBytesTest extends AckGenericTest {
+        @Override
+        public void isLimitBytes(ResultCodes.AckCode obj) {
+            assertTrue(obj.isLimitBytes());
+        }
+    }
+
+    public static class StorageFailureTest extends AckGenericTest {
+        @Override
+        public void isStorageFailure(ResultCodes.AckCode obj) {
+            assertTrue(obj.isStorageFailure());
+        }
+    }
+
+    @Test
+    public void successTest() {
```

Comment:
Public?

Reply:
yeah, didn't notice while looking through the changes.